component/metrics/receiver.go

```diff
@@ -13,9 +14,35 @@ type Receiver struct {
 // RiverCapsule marks receivers as a capsule.
 func (r Receiver) RiverCapsule() {}
 
-// FlowMetric is a wrapper around a single metric without the timestamp
+// FlowMetric is a wrapper around a single metric without the timestamp, FlowMetric is *mostly* immutable
+// the globalrefid can only be set if it is 0
 type FlowMetric struct {
-	GlobalRefID uint64
-	Labels      labels.Labels
-	Value       float64
+	globalRefID *atomic.Uint64
```

Comment:
This represents a single in-flight sample, right? I'm not sure I understand when you would change this in-flight.

Reply:
It's an escape hatch to ensure items have a global id set for the remote_write. It might not be needed, but ensures any metrics have an id assigned.